File: dev/test_data/cpp/test_main/aas_core_meta.v3/input/snippets/verification/data_specification_IEC_61360s_have_value.cpp
bool DataSpecificationIec61360sHaveValue(
  const std::vector<
    std::shared_ptr<types::IEmbeddedDataSpecification>
  >& embedded_data_specifications
) {
  for (const auto& embedded_data_specification : embedded_data_specifications) {
    const std::shared_ptr<
      types::IDataSpecificationContent
    >& content(
      embedded_data_specification->data_specification_content()
    );

    // NOTE (mristin):
	// We need to use dynamic cast due to virtual inheritance. Otherwise,
    // we would have used the model_type followed by a static cast.
	const types::IDataSpecificationIec61360* iec61360 = dynamic_cast<
	  const types::IDataSpecificationIec61360*
	>(content.get());

    if (iec61360 != nullptr) {
	  const common::optional<std::wstring>& maybe_value(
        iec61360->value()
      );

      if (!maybe_value.has_value()) {
        return false;
      }
    }
  }

  return true;
}

File: dev/test_data/golang/test_main/aas_core_meta.v3/input/snippets/Verification/data_specification_IEC_61360s_have_value.go
// Check that [aastypes.DataSpecificationIec61360.Value]
// is defined for all data specifications whose content is given as
// IEC 61360.
func DataSpecificationIEC61360sHaveValue(
	embeddedDataSpecifications []aastypes.IEmbeddedDataSpecification) bool {
	for _, eds := range embeddedDataSpecifications {
		content := eds.DataSpecificationContent()

		ok := aastypes.IsDataSpecificationIEC61360(content)
		if !ok {
			continue
		}
		iec61360 := content.(aastypes.IDataSpecificationIEC61360)

		v := iec61360.Value()
		if v == nil {
			return false
		}
	}
	return true
}
